File: Fase 2/Evidencias Proyecto/Evidencias de sistema/InventaProAPI/Models/Distribuidor.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace InventaProAPI.Models
{
    [Table("Distribuidores")]
    public class Distribuidor
    {
        [Key]
        [Column("distribuidor_id")]
        public int DistribuidorId { get; set; }

        [Required]
        [StringLength(255)]
        public string Nombre { get; set; }

        [StringLength(255)]
        public string? Direccion { get; set; }

        [StringLength(20)]
        public string? Telefono { get; set; }

        [StringLength(255)]
        [Column("correo_electronico")]
        public string? CorreoElectronico { get; set; }

        [Required]
        public int Estado { get; set; }

        [Column("fecha_registro")]
        public DateTime? FechaRegistro { get; set; }

        [Column("fecha_modificacion")]
        public DateTime? FechaModificacion { get; set; }
    }
}

File: Fase 3/Evidencias Grupales/Evidencias Proyecto/Evidencias de sistema/InventaPro/src/app/interfaces/distribuidor.ts
export interface Distribuidor {
  distribuidorId: number;
  nombre: string;
  direccion?: string;
  telefono?: string;
  correoElectronico?: string;
  estado: number;
  fechaRegistro?: string;
  fechaModificacion?: string;

  estadoDesc?: string;
}

export interface DistribuidorCU {
  distribuidorId: number;
  nombre: string;
  direccion?: string;
  telefono?: string;
  correoElectronico?: string;
}